recipes/alembic/all/conanfile.py

```diff
@@ -0,0 +1,92 @@
+from conans import ConanFile, CMake, tools
+import os
+import shutil
+
+
+class AlembicConan(ConanFile):
+    name = "alembic"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/alembic/alembic"
+    description = "Open framework for storing and sharing scene data."
+    topics = ("usb", "scene")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_hdf5": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_hdf5": False
+    }
+    generators = "cmake_find_package"
+    exports_sources = "AlembicIlmBase.cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("openexr/2.5.7")
+        if self.options.with_hdf5:
+            self.requires("hdf5/1.12.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        shutil.copy(os.path.join(self.source_folder, "AlembicIlmBase.cmake"), os.path.join(self._source_subfolder, "cmake"))
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "lib", "Alembic", "CMakeLists.txt"),
+            "RUNTIME DESTINATION ${ALEMBIC_LIB_INSTALL_DIR}",
+            "RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}")
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "lib", "Alembic", "CMakeLists.txt"),
+            "${ALEMBIC_LIB_INSTALL_DIR}",
+            "${CMAKE_INSTALL_LIBDIR}")
```

Comment:
Please move all patching to `build` (eventually in a submethod called `_patch_sources`).
We'ld like to keep the sources as "pristine" as possible.
```suggestion
        tools.replace_in_file(
            os.path.join(self._source_subfolder, "lib", "Alembic", "CMakeLists.txt"),
            "RUNTIME DESTINATION ${ALEMBIC_LIB_INSTALL_DIR}",
            "RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}")
        tools.replace_in_file(
            os.path.join(self._source_subfolder, "lib", "Alembic", "CMakeLists.txt"),
            "${ALEMBIC_LIB_INSTALL_DIR}",
            "${CMAKE_INSTALL_LIBDIR}")
```